.infrastructure/ingress/ingress.yml

```diff
@@ -0,0 +1,18 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+    name: todoapp-ingress
+    namespace: todoapp
+    annotations:
+      nginx.ingress.kubernetes.io/rewrite-target: /$2
+spec:
+  rules:
+  - http:
+      paths:
+      - pathType: ImplementationSpecific
+        path: /(|$)(.*)
```

Comment:
The path regex `/(|$)(.*)` is quite broad and may match more paths than intended. Double-check to ensure this is the desired behavior for your application routing.